test/channelz_test.go

```diff
@@ -202,6 +202,49 @@ func (s) TestCZGetServer(t *testing.T) {
 	}
 }
 
+func (s) TestCZGetSocket(t *testing.T) {
+	e := tcpClearRREnv
+	te := newTest(t, e)
+	lis := te.listenAndServe(&testServer{security: e.security}, net.Listen)
+	defer te.tearDown()
+
+	if err := verifyResultWithDelay(func() (bool, error) {
+		ss, _ := channelz.GetServers(0, 0)
+		if len(ss) != 1 {
+			return false, fmt.Errorf("there should only be one server, not %d", len(ss))
+		}
+
+		serverID := ss[0].ID
+		srv := channelz.GetServer(serverID)
+		if srv == nil {
+			return false, fmt.Errorf("server %d does not exist", serverID)
+		}
+		if srv.ID != serverID {
+			return false, fmt.Errorf("server want id %d, but got %d", serverID, srv.ID)
+		}
+
+		skts := srv.ListenSockets()
+		if got, want := len(skts), 1; got != want {
+			return false, fmt.Errorf("server has %v sockets; want %v", got, want)
```

Comment:
Optional nit: keep consistent with go style guide and also I prefer consistent format across errors (and our codebase): https://g3doc.corp.google.com/go/g3doc/style/decisions.md?cl=head#got-before-want